File: MiniDOMWebCtrls/DMD/Forms/Charts/CChartGrid.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Forms


Namespace Forms
 
 
    Public Class CChartGrid
        Inherits CChartElement

        Private m_ShowSecondaryGrid As Boolean = False
        Private m_ShowXGrid As Boolean = True
        Private m_ShowYGrid As Boolean = True
        
        Public Sub New()
        End Sub

        Public Sub New(ByVal chart As CChart)
            Me.New()
            Me.SetChart(chart)
        End Sub
          
        Public Property ShowXGrid As Boolean
            Get
                Return Me.m_ShowXGrid
            End Get
            Set(value As Boolean)
                If (Me.m_ShowXGrid = value) Then Exit Property
                Me.m_ShowXGrid = value
                Me.InvalidateChart()
            End Set
        End Property

        Public Property ShowYGrid As Boolean
            Get
                Return Me.m_ShowYGrid
            End Get
            Set(value As Boolean)
                If (Me.m_ShowYGrid = value) Then Exit Property
                Me.m_ShowYGrid = value
                Me.InvalidateChart()
            End Set
        End Property

        Public Property ShowSecondaryGrid As Boolean
            Get
                Return Me.m_ShowSecondaryGrid
            End Get
            Set(value As Boolean)
                If (Me.m_ShowSecondaryGrid = value) Then Return
                Me.m_ShowSecondaryGrid = value
                Me.InvalidateChart()
            End Set
        End Property

        Protected Overrides Sub XMLSerialize(writer As XML.XMLWriter)
            writer.WriteAttribute("ShowXGrid", Me.m_ShowXGrid)
            writer.WriteAttribute("ShowYGrid", Me.m_ShowYGrid)
            writer.WriteAttribute("ShowSecondaryGrid", Me.m_ShowSecondaryGrid)
            MyBase.XMLSerialize(writer)
        End Sub

        Protected Overrides Sub SetFieldInternal(fieldName As String, fieldValue As Object)
            Select Case fieldName
                Case "ShowXGrid" : Me.m_ShowXGrid = XML.Utils.Serializer.DeserializeBoolean(fieldValue)
                Case "ShowYGrid" : Me.m_ShowYGrid = XML.Utils.Serializer.DeserializeBoolean(fieldValue)
                Case "ShowSecondaryGrid" : Me.m_ShowSecondaryGrid = XML.Utils.Serializer.DeserializeBoolean(fieldValue)
                Case Else : MyBase.SetFieldInternal(fieldName, fieldValue)
            End Select

        End Sub
    End Class

  

End Namespace

File: MiniDOMWebCtrls/DMD/Forms/Charts/CChartGrid.cs
using System;
using DMD;
using DMD.XML;
using minidom;
using static minidom.Sistema;

namespace minidom.Forms
{
    public class CChartGrid 
        : CChartElement
    {
        private bool m_ShowSecondaryGrid = false;
        private bool m_ShowXGrid = true;
        private bool m_ShowYGrid = true;

        public CChartGrid()
        {
        }

        public CChartGrid(CChart chart) : this()
        {
            SetChart(chart);
        }

        public bool ShowXGrid
        {
            get
            {
                return m_ShowXGrid;
            }

            set
            {
                if (m_ShowXGrid == value)
                    return;
                m_ShowXGrid = value;
                InvalidateChart();
            }
        }

        public bool ShowYGrid
        {
            get
            {
                return m_ShowYGrid;
            }

            set
            {
                if (m_ShowYGrid == value)
                    return;
                m_ShowYGrid = value;
                InvalidateChart();
            }
        }

        public bool ShowSecondaryGrid
        {
            get
            {
                return m_ShowSecondaryGrid;
            }

            set
            {
                if (m_ShowSecondaryGrid == value)
                    return;
                m_ShowSecondaryGrid = value;
                InvalidateChart();
            }
        }

        protected override void XMLSerialize(XMLWriter writer)
        {
            writer.WriteAttribute("ShowXGrid", m_ShowXGrid);
            writer.WriteAttribute("ShowYGrid", m_ShowYGrid);
            writer.WriteAttribute("ShowSecondaryGrid", m_ShowSecondaryGrid);
            base.XMLSerialize(writer);
        }

        protected override void SetFieldInternal(string fieldName, object fieldValue)
        {
            switch (fieldName ?? "")
            {
                case "ShowXGrid":
                    {
                        m_ShowXGrid = DMD.XML.Utils.Serializer.DeserializeBoolean(fieldValue) == true;
                        break;
                    }

                case "ShowYGrid":
                    {
                        m_ShowYGrid = DMD.XML.Utils.Serializer.DeserializeBoolean(fieldValue) == true;
                        break;
                    }

                case "ShowSecondaryGrid":
                    {
                        m_ShowSecondaryGrid = DMD.XML.Utils.Serializer.DeserializeBoolean(fieldValue) == true;
                        break;
                    }

                default:
                    {
                        base.SetFieldInternal(fieldName, fieldValue);
                        break;
                    }
            }
        }
    }
}